src/table/sortable-head-cell.js

```diff
@@ -28,25 +28,33 @@ function SortDirectionIcon({direction}: {direction: SortDirectionT}) {
 }
 
 export default function SortableHeadCell(props: HeadCellPropsT) {
-  const {overrides = {}} = props;
+  const {overrides = {}, extendClick} = props;
 
   const [HeadCell, headCellProps] = getOverrides(
     overrides.HeadCell,
     StyledHeadCell,
   );
+
   const [SortableLabel, sortableLabelProps] = getOverrides(
     overrides.SortableLabel,
     StyledSortableLabel,
   );
 
+  const onClick = () => {
+    props.onSort && props.onSort();
+  };
+
   return (
-    <HeadCell role="columnheader" {...headCellProps}>
+    <HeadCell
+      role="columnheader"
+      {...headCellProps}
+      $cursor={extendClick ? 'pointer' : undefined}
+      onClick={extendClick ? onClick : undefined}
+    >
       <SortableLabel
         aria-label={`sorts table by ${props.title} column`}
         disabled={props.disabled}
-        onClick={() => {
-          props.onSort && props.onSort();
-        }}
+        onClick={!extendClick ? onClick : undefined}
```

Comment:
please in-line the function and check for clarity

Reply:
```
onClick={() => {
  if (props.extendClick && props.onSort) {
    props.onSort();
  }
}}
```

Reply:
Ah just noticed that `onClick` function is used twice. Here and on L52. This is fine